spec/support/custom_matchers/be_same_time_as.rb

```diff
@@ -1,5 +1,13 @@
 RSpec::Matchers.define :be_same_time_as do |expected|
+  regexp = /(.*_spec\.rb:\d+)/
+  called_from = caller.detect { |line| line =~ regexp }
+
   match do |actual|
+    puts <<-MESSAGE
+\rWARNING: The `be_same_time_as` matcher is deprecated and will be removed shortly.
+Use the `be_within` matcher instead: `be_same_time_as(expected_time).precision(1) == be_within(0.1).of(expected_time)`
+#{"Called from " + called_from if called_from}
```

Comment:
Is there a way that called_from *won't* exist?

Reply:
Yes....ready?

RSpec's `shared_examples` don't actually have the spec file they are called from in the backtrace whatsoever because ¯\\\(°_o)/¯ some metaprogramming magic I don't care to find as I know only one place it's used (Embedded Ansible).